tempo_embeddings/embeddings/vector_database.py

```diff
@@ -0,0 +1,336 @@
+                 embedder_name: str = "sentence-transformers/all-MiniLM-L6-v2",
+                 embedder_config: Dict[str,Any] = None,
+                 batch_size: int = 8):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name)
+        self.client =  None
+        self.requires_explicit_embeddings = False
+        if self.embedder_config and self.embedder_config.get("type") == "hf":
+            self.embedding_function = embedding_functions.HuggingFaceEmbeddingFunction(
+                api_key=self.embedder_config.get("api_key"),
+                model_name=embedder_name
+            )
+        elif self.embedder_config and self.embedder_config.get("type") == "custom":
+            self.requires_explicit_embeddings = True
+            self.embedding_function = None
+        else:
+            self.embedding_function = None
+
+        self.config = {
+            "embedder_name": self.embedder_name,
+            "embedder_type": self.embedder_config["type"],
+            "requires_explicit_embeddings": self.requires_explicit_embeddings,
+            "existing_collections": []
+        }
+    
+    def _save_config(self):
+        if self.config:
+            json.dump(self.config, open(f"{self.db_path}/config.json", "w"), indent=2)
+
+    def connect(self):
+        if not self.client:
+            self.client = chromadb.PersistentClient(path=self.db_path)
+            # If the given path existed then we will load the records AND override the parameters with the config
+            config_path = f"{self.db_path}/config.json"
+            if os.path.exists(config_path):
+                self.config = json.load(open(config_path))
+                print(f"Database Path '{self.db_path}' already exists. Loading current configuration\n{self.config}")
+            else:
+                print(f"Creating NEW Database in {self.db_path}...")
+                self._save_config()
+        else:
+            print(f"A connection to the client already exsists in this session: {self.client}")
+
+    def create_new_collection(self, name: str, passages: List[Passage] = None, embeddings: ArrayLike = None) -> Optional[Collection]:
+        if not self.client:
+            print("Please connect to a valid database first")
+            return None
+        # Create NEW collection and Embeds the given passages. Do nothing otherwise
+        try:
+            if self.embedding_function:
+                collection = self.client.create_collection(name, 
+                                                            embedding_function=self.embedding_function,
+                                                            metadata={"hnsw:space": "cosine"})
+            else:
+                collection = self.client.create_collection(name, metadata={"hnsw:space": "cosine"})
```

Comment:
Honestly, I find this part a bit hard to parse. Is the `if` block necessary at all, or could we just pass `embedding_function=None`? In that case, we can just do the same in both cases:
```suggestion
            collection = self.client.create_collection(name, 
                                                            embedding_function=self.embedding_function,
                                                            metadata={"hnsw:space": "cosine"})
```
If not, I would suggest storing the default value for `metadata` in a class field so that it can be updated in a single place in the future.